### ISSUE TITLE
Object Spread - Parsing error: Unexpected token ..

### ISSUE THREAD

**Issue by kentcdodds** [eslint/eslint#4052]
Given this code:

```javascript
module.exports = foo;

function foo(defaultsObject, defaultsArray) {
  const bar = [1,2,3, ...defaultsArray];
  return {bar, ...defaultsObject};
}
```

which compiles with babel-core version 5.8.25 with no special flags/arguments/options

And this eslint config:

```javascript
{
  "env": {
    "es6": true
  }
}
```

I get this error:

> /Users/kentcdodds/Desktop/eslint-spread-object/test.js
>   4:24  error  Parsing error: Unexpected token ..

With eslint version 1.6.0.

Let me know if you need more information. Thanks!

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by michaelficarra** [comment]
That's not an ES6 feature.

**Comment by kentcdodds** [comment]
Oh, is it ES2016?

**Comment by btmills** [comment]
We should have experimental support for this even though it's ES2016, try enabling `experimentalObjectRestSpread` in `ecmaFeatures`.

**Comment by kentcdodds** [comment]
That did it. Thanks! Sorry to bother!

**Comment by abhijeetNmishra** [comment]
:+1:

**Comment by markoshust** [comment]
For anyone stumbling on this, I prefer this method (think it's fairly newer):

    npm install --save-dev babel-eslint

Then add `"parser": "babel-eslint"` to `eslintConfig` in package.json.
